http_handlers.go

```diff
@@ -102,6 +141,113 @@ func getMetadata(filter string) http.HandlerFunc {
 	}
 }
 
+func ec2Handler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	logger.Debug("Calling ec2Handler ")
+	userIP := getIPFromRequest(r)
+	if userIP == "" {
+		return
+	}
+
+	metrics.MetadataRequests.Inc()
+	logger.With("userIP", userIP).Info("Actual IP is: ")
+	hw, err := getByIP(context.Background(), hegelServer, userIP) // returns hardware data as []byte
+	if err != nil {
+		metrics.Errors.WithLabelValues("metadata", "lookup").Inc()
+		logger.Info("Error in finding hardware: ", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	res, err := processEC2Query(r.URL.Path)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("404 not found"))
+		if err != nil {
+			logger.Error(err, "failed to write error response")
+		}
+	}
+
+	var resp []byte
+	if filter, ok := res.(string); ok {
+		resp, err = filterMetadata(hw, filter)
+		if err != nil {
+			logger.Error(err, "Error in filtering metadata: ")
+		}
+	} else if submenu, ok := res.(map[string]interface{}); ok {
+		var keys []string
+		for item := range submenu {
+			switch item {
+			case "_base": // _base is only used to keep track of the base filter, not a metadata item
+				continue
+			case "spot": // list only if instance is spot
+				spotFilter := fmt.Sprint(submenu["_base"], submenu[item].(map[string]interface{})["_base"])
```

Comment:
we don't need any of these `fmt.Sprint` calls over `arg1 + arg2` since they are both strings already.

Reply:
I'm not sure if this makes a difference, but `submenu["_base"]` is actually an `interface{}` so if I use the `+` to concat, I would need to do type assertions:
```
spotFilter := submenu["_base"].(string) + submenu[item].(map[string]interface{})["_base"].(string)
```
which looked really cluttered to me. 
I guess, how does `sprint` compare to type assertions in terms of efficiency?

Reply:
fixed: https://github.com/tinkerbell/hegel/pull/22/commits/af31a740d81ab418184b58714a13d1118e059c87#diff-417a3d9f2a96ef7dc446d080e58f03b1R186